delft3dworker/models.py

```diff
@@ -479,11 +490,160 @@ class Container(models.Model):
     # docker container ids are sha256 hashes
     docker_id = models.CharField(max_length=64, blank=True, default='')
 
-    def _update_state_and_save(self):
+    docker_log = models.TextField(blank=True, default='')
 
-        # TODO: write _update_state_and_save method
+    def update_task_result(self):
+        """
+        This method will get the result from the last task it executed, given
+        that there is a result. If the task is not ready, this will not do
+        anything.
+        """
+
+        if self.task_uuid is None:
+            return
+
+        result = AsyncResult(id=self.task_uuid)
+
+        if result.ready():
```

Comment:
I'd say neither: the task is done - update the Container fields. 

If anything failed, let the `desired_state` and `docker_state` take care of things.